### DISCUSSION TITLE
Modifying Database Data in OpenWPM during Each Command Execution without Changing Core Code

### DISCUSSION THREAD

**Discussion by chavoshimahla79** [openwpm/OpenWPM#1064] [Q&A] [answered]
Hi,

i have a specific requirement for my project involving the modification of database data within OpenWPM. My goal is to achieve this without the need to alter the core code of OpenWPM itself.

In my use case, I need to perform data manipulation, such as modification or deletion, during the execution of a command, particularly within the execute function. While I am aware that OpenWPM provides the DataSocket.store_record functionality for storing data, I also need the ability to perform these data modifications as part of my workflow.

My aim is to develop on top of OpenWPM without making direct changes to its core codebase. Hence, I am reaching out to inquire if there is an API or approach that would allow me to manipulate the database data during each run without the need for modifications to the OpenWPM source code (like extension and etc).

**Comment by vringar** [comment, marked as answer]
Hey,
this is currently unsupported.
My initial approach would be to implement a custom StructuredStorageController, that writes through everything as normal except for writes to a `my_special_command_table` and then data for that special table would contain the deletions or modifications you want to do.
